### PR TITLE
[Homework #6] - JS-Part 1 by Pavel Didenko

### PR DESCRIPTION
**Pull request by Win4esterD**
Solved all the tasks except №16. The 16th makes my brain melt.

### CODE REVIEW

File: homeworks/pavel-didenko/06-js-basics-1/task6.js [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    return "Escaped!";
+  }
+}
+
+
+
+//task15: https://www.codewars.com/kata/duplicate-encoder
+
+function duplicateEncode(word){
+  word = word.toLowerCase();
```
Comment (ilya-ischenko) [inline homeworks/pavel-didenko/06-js-basics-1/task6.js:250]: Do not mutate input property

**Review by ilya-ischenko — COMMENTED** [review]
In root folder of our laba we have "package.json" file with needed dependecies, make "yarn" command in root directory in order to install packages and format your code, please.

File: homeworks/pavel-didenko/06-js-basics-1/task6.js [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+function rentalCarCost(d) {
+  if(d >= 7){
+    return 40 * d - 50;
+  }else if(d >= 3){
+    return d * 40 - 20;
+  }else{
+    return d * 40;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Win4esterD) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: image

Is it fine to leave "7" and "3" in the "if/else" block for this case? It feels a bit strange to call variables like:
const daysInWeek = 7;
const daysForFirstDiscount = 3;

At least, when I already have variables like "sevenDaysDiscount" and "ThreeDaysDiscount".

File: homeworks/pavel-didenko/06-js-basics-1/task6.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  return counter;
+}
+
+function findOdd(arr) {
+  for(let i = 0; i < arr.length; i++){
+    if(countNumbersInArray(arr[i], arr) % 2 > 0){
+      return arr[i];
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (Win4esterD) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: Nice idea, now the code looks this way:

image

File: homeworks/pavel-didenko/06-js-basics-1/task6.js [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+//task11: https://www.codewars.com/kata/zipwith
+
+function zipWith(fn, arr1, arr2) {
+  const result = [];
+
+  if(arr1.length <= arr2.length){
+    arr1.map((item, index) => result.push(fn(item, arr2[index])));
+  }else{
+    arr2.map((item, index) => result.push(fn(arr1[index], item)));
+  }
```
[the remark this answers is not in the document]
Comment (Win4esterD) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: Ah, yeah, this was the funny one. Resolved. :)

File: homeworks/pavel-didenko/06-js-basics-1/task6.js [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+  let result = '';
+
+  for(let i in value){
+    if(!isNaN(parseInt(value[i]))){
+      result += value[i].toString();
+    }
+  }
+
+  return parseInt(result);
+}
```
[the remark this answers is not in the document]
Comment (Win4esterD) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: Fine.

File: homeworks/pavel-didenko/06-js-basics-1/task6.js (hunk 1 above)
Comment (Win4esterD) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: Hmmm, at least it passed the tests. OK, no problem, I hope your trick with object from the conversation above should work better. ;)

image

File: homeworks/pavel-didenko/06-js-basics-1/task6.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilya-ischenko) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: I think 3 and 7 is ok in this case

File: homeworks/pavel-didenko/06-js-basics-1/task6.js (hunk 1 above)
Comment (ilya-ischenko) [reply homeworks/pavel-didenko/06-js-basics-1/task6.js]: It's bad practice to mutate input, try to avoid it

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**